File: FindBug.java
import java.io.*;
import java.nio.file.*;
import java.util.*;

public class FindBug {
    public static void main(String[] args) throws Exception {
        Path dir = Paths.get("d:/ai/MocPlugin/src/main/java/me/user/moc/ability/impl");
        Files.walk(dir).filter(p -> p.toString().endsWith(".java")).forEach(p -> {
            try {
                String content = new String(Files.readAllBytes(p), "UTF-8");
                if (content.contains("PlayerInteractEvent") && content.contains("checkCooldown(")) {
                    boolean noRight = !content.contains("Action.RIGHT") && !content.contains("contains(\"RIGHT\")");
                    String noSpace = content.replaceAll("\\s+", "");
                    boolean noItemNull = !noSpace.contains("item==null") && !noSpace.contains("getItem()==null");
                    if (noRight || noItemNull) {
                        System.out.println("Suspect: " + p.getFileName() + " | NoRight: " + noRight + " | NoItemNull: "
                                + (noRight ? false : noItemNull));
                    }
                }
            } catch (Exception e) {
            }
        });
    }
}

File: find_bug.py
import os
import re

base_dir = r"d:\ai\MocPlugin\src\main\java\me\user\moc\ability\impl"
files = [os.path.join(base_dir, f) for f in os.listdir(base_dir) if f.endswith(".java")]

print("=== Suspect (No Action check) ===")
for fpath in files:
    with open(fpath, "r", encoding="utf-8") as f:
        content = f.read()
    if "PlayerInteractEvent" in content and "checkCooldown(" in content:
        if "Action.RIGHT_CLICK" not in content and "contains(\"RIGHT\")" not in content and "isSneaking" not in content:
            print(os.path.basename(fpath))

print("=== Suspect (No item==null check) ===")
for fpath in files:
    with open(fpath, "r", encoding="utf-8") as f:
        content = f.read()
    if "PlayerInteractEvent" in content and "checkCooldown(" in content:
        c = content.replace(" ", "").replace("\n", "")
        if "getItem()==null" not in c and "item==null" not in c:
            print(os.path.basename(fpath))
